app/actions/remote/search.ts

```diff
@@ -129,10 +131,15 @@ export const searchPosts = async (serverUrl: string, teamId: string, params: Pos
     }
 };
 
-export const searchFiles = async (serverUrl: string, teamId: string, params: FileSearchParams): Promise<{files?: FileInfo[]; channels?: string[]; error?: unknown}> => {
+export const searchFiles = async (serverUrl: string, teamId: string, params: FileSearchParams, channel?: ChannelModel): Promise<{files?: FileInfo[]; channels?: string[]; error?: unknown}> => {
```

Comment:
```suggestion
export const searchFiles = async (serverUrl: string, channel: ChannelModel, params: FileSearchParams): Promise<{files?: FileInfo[]; channels?: string[]; error?: unknown}> => {
```

Seems to be a better option

Reply:
@enahum `searchFiles` function has been used in multiple places, for example, https://github.com/mattermost/mattermost-mobile/blob/e7fe7c23ffcf6844a35630594864b1548a2fbdda/app/screens/home/search/search.tsx#L226
here. We have to fetch the channel object, which will be an extra db call. Rather than that, this is the best option. Through?